### PR TITLE
TreeHeightViewModelTests

### PR DESCRIPTION
**Pull request by Mzazi25**
Added unit tests for selecting colors on the TreeHeightViewModel.

### CODE REVIEW

File: app/src/test/java/org/greenstand/android/TreeTracker/utils/FakeFileGenerator.kt [hunk 1]
```diff
@@ -191,4 +194,37 @@ object FakeFileGenerator {
...
+        shadowColor = AppColors.YellowShadow,
+        disabledColor = AppColors.GrayShadow,
+        disabledShadowColor = AppColors.GrayShadow
+    )
+    val fakeSkyBlue = DepthButtonColors(
+        color = AppColors.SkyBlue,
+        shadowColor = AppColors.SkyBlueShadow,
+        disabledColor = AppColors.GrayShadow,
+        disabledShadowColor = AppColors.GrayShadow
+    )
```
Comment (Elforama) [inline app/src/test/java/org/greenstand/android/TreeTracker/utils/FakeFileGenerator.kt:229]: There is no need to make test colors. Just use the AppButtonColors class.

File: app/src/test/java/org/greenstand/android/TreeTracker/treeheight/TreeHeightViewModelTest.kt [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+        val result = treeHeightSelectionViewModel.state.getOrAwaitValueTest().selectedColour
+        Assert.assertEquals(result, FakeFileGenerator.fakeYellow)
+    }
+
+    @Test
+    fun `selected color is message purple, assert correct color`()= runBlocking {
+        treeHeightSelectionViewModel.selectColor(AppButtonColors.MessagePurple)
+        val result = treeHeightSelectionViewModel.state.getOrAwaitValueTest().selectedColour
+        Assert.assertEquals(result, FakeFileGenerator.fakeMessagePurple)
+    }
```
Comment (Elforama) [inline app/src/test/java/org/greenstand/android/TreeTracker/treeheight/TreeHeightViewModelTest.kt:68]: All these tests test the same thing. We only need the first one.

**Review by Elforama — CHANGES_REQUESTED** [review]
Please address the comments. Also, we should usually always be testing the interactions with the mocked classes. In this case, testing that the TreeCapturer gets a color set to it is an important thing to test as well.

File: app/src/test/java/org/greenstand/android/TreeTracker/utils/FakeFileGenerator.kt (hunk 1 above)
Comment (Mzazi25) [reply app/src/test/java/org/greenstand/android/TreeTracker/utils/FakeFileGenerator.kt]: done

File: app/src/test/java/org/greenstand/android/TreeTracker/treeheight/TreeHeightViewModelTest.kt [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+    var instantTaskExecutorRule = InstantTaskExecutorRule()
+
+    @get:Rule
+    var mainCoroutineRule = MainCoroutineRule()
+
+    private val treeCapturer = mockk<TreeCapturer>(relaxed = true)
+    private lateinit var treeHeightSelectionViewModel: TreeHeightSelectionViewModel
+
+    @Before
+    fun setupViewModel(){
```
[the remark this answers is not in the document]
Comment (Mzazi25) [reply app/src/test/java/org/greenstand/android/TreeTracker/treeheight/TreeHeightViewModelTest.kt]: okay

**Review by Elforama — CHANGES_REQUESTED** [review]

File: app/src/test/java/org/greenstand/android/TreeTracker/treeheight/TreeHeightViewModelTest.kt [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+    @Before
+    fun setup(){
+        treeHeightSelectionViewModel = TreeHeightSelectionViewModel(treeCapturer)
+    }
+
+    @Test
+    fun `selected color is progress green, asserts correct color added `()= runBlocking {
+        val color = AppButtonColors.ProgressGreen
+        treeHeightSelectionViewModel.selectColor(color)
+        treeCapturer.addAttribute(Tree.TREE_COLOR_ATTR_KEY, color.toString())
```
[the remark this answers is not in the document]
Comment (Elforama) [reply app/src/test/java/org/greenstand/android/TreeTracker/treeheight/TreeHeightViewModelTest.kt:39]: You are now verifying that it's used, but your manually triggering its usage. We want to verify its usage inside the class we're testing, so there is no need to trigger it yourself.

**Review by Elforama — APPROVED** [review]

**Comment by Elforama** [comment]
Please update this with the latest master branch and I can merge it.

**Pull request merged**